pkg/receiver/gitproviderreceiver/internal/scraper/githubscraper/github_scraper.go

```diff
@@ -330,8 +259,40 @@ func (ghs *githubScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
 					}
 				}
 			}
+			var prCursor *string
+			var pullRequests []PullRequestNode
+			var perPage int = 100
+
+			prCount, err := getPullRequestCount(ctx, genClient, name, ghs.cfg.GitHubOrg)
+			if err != nil {
+				ghs.logger.Sugar().Errorf("error getting pull request count", zap.Error(err))
+			}
+			ghs.logger.Sugar().Debugf("pull request count: %v for repo %v", prCount, repo)
 
-			ghs.getRepoPullRequestInformation(repoInfo)
+			prPages := getNumPages(float64(perPage), float64(prCount.Repository.PullRequests.TotalCount))
+			ghs.logger.Sugar().Debugf("pull request pages: %v for repo %v", prPages, repo)
+
+			for i := 0; i < prPages; i++ {
+				pr, err := getPullRequestData(ctx, genClient, name, ghs.cfg.GitHubOrg, perPage, prCursor)
+				if err != nil {
+					ghs.logger.Sugar().Errorf("error getting pull request data", zap.Error(err))
+				}
+
+				pullRequests = append(pullRequests, pr.Repository.PullRequests.Nodes...)
+
+				prCursor = &pr.Repository.PullRequests.PageInfo.EndCursor
+			}
+
+			for _, pr := range pullRequests {
+				createTime := pr.CreatedAt
+				closedTime := time.Now()
```

Comment:
I'm thinking through this and the closed time, if the pr isn't closed, couldn't be now. if it is closed, then the closed time also wouldn't be now. If we want to subtract the current time from the creation time, then we could use `now` but what if closed date actually comes from the pull request data? Also, should we only look at open pr's or all prs (open & cosed)